build/gen_stub.php

```diff
@@ -323,6 +483,40 @@ public function toArginfoType(): ArginfoType {
         return new ArginfoType($classTypes, $builtinTypes);
     }
 
+    public function toOptimizerTypeMask(): string {
+        $optimizerTypes = [];
+
+        foreach ($this->types as $type) {
+            if ($type->isBuiltin) {
```

Comment:
Why do we need to check isBuiltin here?

Reply:
That's because only built-in types have a type mask (there is an `assert($this->isBuiltin)` in `SimpleType::toTypeMask()`). However, you are right, it would be better to add a special case in `SimpleType::toOptimizerTypeMask()` indeed, just like how I did for `mixed` et al.

Reply:
It looks like you adjusted the toOptimizerTypeMask implementation but still left the check here.